Feature/revive lunr

dlqs opened the pull request:
This PR brings back Lunr client side search as the offline search. The index is pre-built during Netlify build and served alongside other static assets. Collection search can be restricted to offline search only through a setting in `_config.yml`.

dlqs committed:
add skeleton config for offline search

dlqs committed:
re-enable lunr search

dlqs committed:
change to  prebuild lunr index

dlqs committed:
lunr pre build index on server side

LoneRifle commented on _config.yml:
```diff
@@ -9,6 +9,7 @@ kramdown:
 repository: opendocsg/opendoc-theme
 server_PROD: https://search.opendoc.sg
 server_DEV: https://search-staging.opendoc.sg
+offline_search_only: true
```
this should be false, as a default. Most of the opendocs should be indexed by ES. I realise you're testing this on netlify, but be mindful to have this corrected before merging.

dlqs replied:
Yup!

dlqs committed:
make document title show up in search

 - however, document titles are whitelisted from search so that searching for a
   term that happens to be in a document title does not clog up the search
   results with all the sections from that document

dlqs committed:
remove unneeded client side lunr index generation

dlqs committed:
refactor

dlqs committed:
add fuzzy search

dlqs committed:
rename snippet to contentsnippet

dlqs committed:
add highlighting to title in search results

 - seciton title only, not document title

dlqs committed:
update lunr to 2.3.6

dlqs committed:
move offline config variables to new file

 - place offline config as second arg to --config

dlqs committed:
further refine the search

dlqs committed:
move back offline setting

dlqs committed:
Merge branch v2 into feature/revive-lunr

dlqs committed:
add build command for building lunr index

dlqs committed:
Merge branch 'v2' of github.com:opendocsg/opendoc-theme into feature/revive-lunr

dlqs committed:
reference lunr in assets

dlqs committed:
fix lunr index url

dlqs committed:
prevent null pointer error while index is loading

dlqs committed:
set default offline search

dlqs committed:
fix online search url

pregnantboy committed:
Code cleanup and contiguous highlighting in search results

pregnantboy committed:
Merge branch 'v2' into feature/revive-lunr

# Conflicts:
#	assets/js/search.js

dlqs committed:
do not generate lunr index when offline is false
